### ISSUE TITLE
[labs/compiler] issue with TypeScript 5.3

### ISSUE THREAD

**Issue by driesva** [lit/lit#4460]
### Which package(s) are affected?

Other/unknown (please mention in description)

### Description

When building our application (with Webpack) we encounter following issue during Lit template compilation after upgrading TypeScript from 5.2.2 to 5.3.3

```
Module parse failed: Identifier 'B_1' has already been declared (22:6)
File was processed with these loaders:
 * ./node_modules/ts-loader/index.js
You may need an additional loader to handle the result of these loaders.
| const { AttributePart: A_1, BooleanAttributePart: B_1 } = litHtmlPrivate_1._$LH;
| const b_1 = i => i;
> const B_1 = { h: b_1 `\n      
```

### Reproduction

We invoke the compiler using Webpack in the following way:

```javascript
      {
        loader: 'ts-loader',
        options: {
          getCustomTransformers: () => ({
            before: [
              compileLitTemplates()
            ]
          })
        }
      }
```

### Workaround

I have not found a workaround

### Is this a regression?

Yes. This used to work, but now it doesn't.

### Affected versions

lit 3.1.0 lit-labs/compiler 1.0.1

### Browser/OS/Node environment

Node: v21.5.0
npm: 10.2.4

**Comment by mdownes** [comment]
I got this to start working again by upgrading to typescript@5.3.3 and using @lit-labs/compiler@1.0.3(which uses 5.3.3 internally)

**Comment by driesva** [comment]
Indeed, that combination seems to be working. (Using the latest TS at this moment, 5.4.5, results again in the same error)

**Comment by suyouwanggang** [comment]
typescript 5.5.4 以上版本，都有这个问题
